pkg/api/v1/clients/memory/resource_client.go

```diff
@@ -197,9 +198,15 @@ func (rc *ResourceClient) Delete(namespace, name string, opts clients.DeleteOpts
 func (rc *ResourceClient) List(namespace string, opts clients.ListOpts) (resources.ResourceList, error) {
 	opts = opts.WithDefaults()
 	cachedResources := rc.cache.List(rc.Prefix(namespace))
+
+	labelSelector, err := kubeutils.ToLabelSelector(opts)
```

Comment:
Should we gate this functionality to only apply it _if_ a labelSelector is non-nil?

Reply:
a04fd4046965b63a9381227c3dece888d20ed542